src/s3/copyobject.cpp

```diff
@@ -19,12 +19,19 @@ boost::asio::awaitable<void> irods::s3::actions::handle_copyobject(
 {
     auto thing = irods::s3::get_connection();
     std::cerr << "I'm about to authenticate for copying?" << std::endl;
-    if (!irods::s3::authentication::authenticates(*thing, parser, url)) {
+
+    auto irods_username = irods::s3::authentication::authenticates(*thing, parser, url);
+    if (!irods_username) {
         beast::http::response<beast::http::empty_body> response;
         response.result(beast::http::status::forbidden);
         beast::http::write(socket, response);
         co_return;
     }
+
+    // Reconnect to the iRODS server as the target user.
+    // The rodsadmin account from the config file will act as the proxy for the user.
+    thing = irods::s3::get_connection(*irods_username);
```

Comment:
Maybe a dumb question...

`irods::s3::authentication::authenticates` returns a `std::optional<std::string>` and `irods::s3::get_connection` takes a `const std::optional<std::string>&`... Can we just pass in `irods_username` instead of accessing the value? Is there a difference?

Reply:
Good observation.

Will change.

Reply:
Done.